<?xml version="1.0"?>
<launch>

  <arg name="rviz" default="0" />
  <arg name="planning" default="0" />
  <arg name="application" default="0" />

  <include file="$(find twc_support)/launch/load_workcell.launch" />

  <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(find twc_application)/config/twc.rviz" output="screen" required="true"/>
  <node if="$(arg planning)" name="planning_manager_node" pkg="twc_motion_planning" type="planning_manager_node" required="true"/>
  <node if="$(arg application)" name="application_node" pkg="twc_application" type="application_node" required="true">
    <param name="tool_path" type="str" value="$(find twc_application)/config/job_path.yaml"/>
  </node>

  <!-- Launch GUI to drive joints   -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

  <node name="tesseract_contact_monitoring" pkg="tesseract_monitoring" type="tesseract_monitoring_contacts_node" required="true">
    <param name="publish_markers" value="true"/>
  </node>
</launch>
